overlord/snapstate/snapstate.go

```diff
@@ -724,6 +725,11 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	} else {
 		installSet.MarkEdge(prepare, LastBeforeLocalModificationsEdge)
 	}
+	if flags&noRestartBoundaries == 0 {
+		if err := SetEssentialSnapsRestartBoundaries(st, nil, []*state.TaskSet{installSet}); err != nil {
```

Comment:
do we need more explicit checks in the unit tests about this having been called?

Reply:
absolutely, added a unit test to cover this